src/contracts/GPv2Settlement.sol

```diff
@@ -190,35 +190,27 @@ contract GPv2Settlement is ReentrancyGuard {
     /// [`computeTradeExecution`] for more details.
     /// @param tokens An array of ERC20 tokens to be traded in the settlement.
     /// @param clearingPrices An array of token clearing prices.
-    /// @param encodedTrades Encoded trades for signed orders.
+    /// @param trades Trades for signed orders.
     /// @return executedTrades Array of executed trades.
     function computeTradeExecutions(
         IERC20[] calldata tokens,
         uint256[] calldata clearingPrices,
-        bytes calldata encodedTrades
+        GPv2Trade.Data[] calldata trades
     ) internal returns (GPv2TradeExecution.Data[] memory executedTrades) {
-        (uint256 tradeCount, bytes calldata remainingEncodedTrades) =
-            encodedTrades.decodeTradeCount();
-        executedTrades = new GPv2TradeExecution.Data[](tradeCount);
-
-        GPv2Encoding.Trade memory trade;
-        uint256 i = 0;
-        while (remainingEncodedTrades.length != 0) {
-            remainingEncodedTrades = remainingEncodedTrades.decodeTrade(
-                domainSeparator,
-                tokens,
-                trade
-            );
+        GPv2Trade.Recovered memory trade = GPv2Trade.newRecovered();
```

Comment:
These are all very good points. I'll come up with something to better distinguish between the kinds of trade data. I liked `Data` because `GPv2Trade.Data` reads so nice, but you are right that here, as we have two kinds of "trades" the names need to be more descriptive.

Reply:
My issue with `Packed` is that it might be confused with `abi.encodePacked` which is not what it is.

Reply:
I moved stuff around (and to me it makes more sense, and will work better once we make `GPv2Signing` a mixin instead of a library). Let me know if this is OK.